src/dotnet/ToolManifest/ToolManifestEditor.cs

```diff
@@ -99,15 +107,67 @@ public ToolManifestEditor(IFileSystem fileSystem = null, IDangerousFileDetector
 
         private SerializableLocalToolsManifest DeserializeLocalToolsManifest(FilePath possibleManifest)
         {
+            var json = _fileSystem.File.ReadAllText(possibleManifest.Value);
+            var serializableLocalToolsManifest = new SerializableLocalToolsManifest();
             try
             {
-                return JsonConvert.DeserializeObject<SerializableLocalToolsManifest>(
-                    _fileSystem.File.ReadAllText(possibleManifest.Value), new JsonSerializerSettings
+                using (JsonDocument doc = JsonDocument.Parse(json))
+                {
+                    JsonElement root = doc.RootElement;
+
+                    if (root.TryGetPropertyValue<int>(JsonPropertyVersion, out var version))
                     {
-                        MissingMemberHandling = MissingMemberHandling.Ignore
-                    });
+                        serializableLocalToolsManifest.Version = version;
+                    }
+
+                    if (root.TryGetPropertyValue<bool>(JsonPropertyIsRoot, out var isRoot))
+                    {
+                        serializableLocalToolsManifest.IsRoot = isRoot;
+                    }
+
+                    if (root.TryGetProperty(JsonPropertyTools, out var tools))
+                    {
+                        serializableLocalToolsManifest.Tools =
+                            new Dictionary<string, SerializableLocalToolSinglePackage>();
+
+                        foreach (var toolJson in tools.EnumerateObject())
+                        {
+                            var serializableLocalToolSinglePackage = new SerializableLocalToolSinglePackage();
+                            if (toolJson.Value.TryGetPropertyValue<string>(JsonPropertyVersion, out var versionJson))
+                            {
+                                serializableLocalToolSinglePackage.Version = versionJson;
+                            }
+
+                            var commands = new List<string>();
+                            if (toolJson.Value.TryGetProperty(JsonPropertyCommands, out var commandsJson))
+                            {
+                                foreach (var command in commandsJson.EnumerateArray())
+                                {
+                                    try
+                                    {
+                                        commands.Add(command.GetString());
```

Comment:
Same as above, a type check on `command` to ensure it's a string, otherwise throw the same exception message (i.e. "expected an array of strings...").